service/matching/taskReader.go

```diff
@@ -301,3 +313,11 @@ func (tr *taskReader) emitTaskLagMetric(ackLevel int64) {
 	maxReadLevel := tr.tlMgr.taskWriter.GetMaxReadLevel()
 	tr.scope().UpdateGauge(metrics.TaskLagPerTaskQueueGauge, float64(maxReadLevel-ackLevel))
 }
+
+func (tr *taskReader) backoff(duration time.Duration) {
+	if tr.backoffTimer == nil {
+		tr.backoffTimer = time.AfterFunc(duration, func() {
```

Comment:
Will make the same change as https://github.com/temporalio/temporal/pull/3195 here.